src/rulesets/Base/sort.jl

```diff
@@ -33,3 +37,54 @@ function rrule(::typeof(sort), xs::AbstractVector; kwargs...)
     end
     return ys, sort_pullback
 end
+
+#####
+##### `sortslices`
+#####
+
+function rrule(::typeof(sortslices), x::AbstractArray{<:Number}; dims::Integer, kw...)
+    p = sortperm(collect(eachslice(x; dims=dims)); kw...)
+    inds = ntuple(d -> d == dims ? p : (:), ndims(x))
+    function sortslices_pullback(dy)
+        # No actual need to zero this, and if you didn't, then you could widen eltype
+        # Also, you could use similar(dy) here not x, same size?
+        dx = _zerolike_writeat(x, unthunk(dy), (), inds...)
+        return (NoTangent(), ProjectTo(x)(dx))
+    end
+    return x[inds...], sortslices_pullback
+end
+
+#####
+##### `unique`
+#####
+
+function rrule(::typeof(unique), x::AbstractArray{<:Number}; dims=:)
+    axes_x = axes(x)
+    y = unique(x; dims=dims)  # accepts only dims=: or dims::Integer
+    function unique_pullback(dy_raw)
+        dy = unthunk(dy_raw)
+        if length(x) == length(y)
+            # Short-circuit for the case of all unique, since `mask` is fairly expensive:
+            dx = reshape(dy, axes_x)
+            return (NoTangent(), ProjectTo(x)(dx))
+        end
+        if dims isa Colon
+            xs, ys = vec(x), y
+        else
+            xs, ys = collect(eachslice(x; dims=dims)), collect(eachslice(y; dims=dims))
+        end
+        mask = isequal.(permutedims(ys), xs)  # unique([0.0, -0.0, NaN, NaN])
+        mask .= (mask .== cumsum(mask, dims=1) .== true)
+        keep = map(I -> I[1], findall(mask))
+        if dims isa Colon
+            # The function `_zerolike_writeat` is defined near `maximum`, allows
+            # second derivatives. Should perhaps eventually be shared with `getindex`.
```

Comment:
```suggestion
            # The function `_zerolike_writeat` allows second derivatives.
            # Should perhaps eventually be shared with `getindex`.
```

The bit about where it is defined confused me, I thought it meant mathematically defined.

Finding where things are defined is a problem for your IDE (or for `methods` in the REPL)

Reply:
Sure

Reply:
Tests pass, will commit this & then merge.

Will bump version with next PR.